html5/kubeinvaders.js

```diff
@@ -142,16 +156,20 @@ var chart_status_code_dict = {
     "Other": 1
 };
 
+async function checkHTTP(url, elementId) {
+    const response = await fetch(url);
+    $("#" + elementId).val(response.status);
+}
 
-function checkHTTP(url, elementId) {
-    var oReq = new XMLHttpRequest();
-    oReq.onreadystatechange = function () {
-        if (this.readyState === XMLHttpRequest.DONE) {
-            $("#" + elementId).val(this.status);
-        }
-    };;
-    oReq.open("GET", url);
-    oReq.send();
+function httpGetToK8S(pathname, queryParams) {
+    const url = new URL(k8s_url);
+    url.pathname = pathname;
+
+    if (queryParams) {
+        Object.keys(queryParams).forEach(key => url.searchParams.set(key, queryParams[key]));
+    }
+
+    return fetch(url);
 }
```

Comment:
Created this helper function to reduce the code needed to send GET requests to `k8s_url`